2-copy-of-code/lesson-17/tests/data/cartTest.js

```diff
@@ -1,37 +1,31 @@
-import {addToCart, cart, loadFromStorage} from '../../data/cart.js';
+import {cart} from '../../data/cart-class.js';
 
 describe('test suite: addToCart', () => {
   it('adds an existing product to the cart', () => {
     spyOn(localStorage, 'setItem');
 
-    spyOn(localStorage, 'getItem').and.callFake(() => {
-      return JSON.stringify([{
-        productId: 'e43638ce-6aa0-4b85-b27f-e1d07eb678c6',
-        quantity: 1,
-        deliveryOptionId: '1'
-      }]);
-    });
-    loadFromStorage();
+    cart.cartItems = [{
+      productId: 'e43638ce-6aa0-4b85-b27f-e1d07eb678c6',
+      quantity: 1,
+      deliveryOptionId: '1'
+    }];
 
-    addToCart('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
-    expect(cart.length).toEqual(1);
+    cart.addToCart('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
+    expect(cart.cartItems.length).toEqual(1);
     expect(localStorage.setItem).toHaveBeenCalledTimes(1);
-    expect(cart[0].productId).toEqual('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
-    expect(cart[0].quantity).toEqual(2);
+    expect(cart.cartItems[0].productId).toEqual('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
+    expect(cart.cartItems[0].quantity).toEqual(2);
   });
 
   it('adds a new product to the cart', () => {
     spyOn(localStorage, 'setItem');
 
-    spyOn(localStorage, 'getItem').and.callFake(() => {
-      return JSON.stringify([]);
-    });
-    loadFromStorage();
+    cart.cartItems = [];
 
-    addToCart('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
-    expect(cart.length).toEqual(1);
+    cart.addToCart('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
+    expect(cart.cartItems.length).toEqual(1);
     expect(localStorage.setItem).toHaveBeenCalledTimes(1);
```

Comment:
In the testing exercise, we made a test with toHaveBeenCalledWith but in this code why isn't there that testing code? However, all other test pass except  toHaveBeenCalledWith test. But toHaveBeenCalledWith test don't pass with this function. What can I do to pass this  toHaveBeenCalledWith test?